packages/jellyfish-transaction/src/script/dftx/dftx_pool.ts

```diff
@@ -41,13 +39,18 @@ export class CPoolSwap extends ComposableBuffer<PoolSwap> {
       ComposableBuffer.single<Script>(() => ps.toScript, v => ps.toScript = v, v => new CScript(v)),
       ComposableBuffer.varUInt(() => ps.toTokenId, v => ps.toTokenId = v),
       {
-        fromBuffer (buffer: SmartBuffer) {
+        fromBuffer: (buffer: SmartBuffer): void => {
           const integer = readBigNumberUInt64(buffer)
           const fraction = readBigNumberUInt64(buffer)
-          ps.maxPrice = { integer, fraction }
+          ps.maxPrice = integer.plus(fraction.dividedBy(ONE_HUNDRED_MILLION))
         },
-        toBuffer (buffer: SmartBuffer) {
-          const { integer, fraction } = ps.maxPrice
+        toBuffer: (buffer: SmartBuffer): void => {
+          const n = ps.maxPrice.multipliedBy(ONE_HUNDRED_MILLION)
```

Comment:
I think it got over complicated here
```suggestion
          const integer = ps.maxPrice.integerValue(BigNumber.ROUND_FLOOR) // need to check the behavior in test
          const fraction = ps.maxPrice.mod(1).times(ONE_HUNDRED_MILLION)
```

Reply:
I don't think it is clearer this way. I dont' think integerValue and mod(1) are intuitive. 
Moreover, the current implementation is coherent with the cpp code one can find in `rpc_poolpair.cpp` :

`CAmount maxPrice = AmountFromValue(metadataObj["maxPrice"]);
  poolSwapMsg.maxPrice.integer = maxPrice / COIN
  poolSwapMsg.maxPrice.fraction = maxPrice % COIN;
`